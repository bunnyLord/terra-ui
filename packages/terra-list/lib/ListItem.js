'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _chevronRight = require('terra-icon/lib/icon/themeable/chevronRight.svg');

var _chevronRight2 = _interopRequireDefault(_chevronRight);

require('../src/list-item.scss');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var propTypes = {
  className: _react.PropTypes.string,
  content: _react.PropTypes.element,
  isSelected: _react.PropTypes.bool,
  isSelectable: _react.PropTypes.bool,
  hasChevron: _react.PropTypes.bool
};

var defaultProps = {
  className: '',
  content: undefined,
  isSelected: false,
  isSelectable: undefined,
  hasChevron: undefined
};

var ListItem = function ListItem(_ref) {
  var className = _ref.className,
      content = _ref.content,
      isSelected = _ref.isSelected,
      isSelectable = _ref.isSelectable,
      hasChevron = _ref.hasChevron,
      customProps = _objectWithoutProperties(_ref, ['className', 'content', 'isSelected', 'isSelectable', 'hasChevron']);

  var listItemClassNames = (0, _classnames2.default)(['terra-ListItem', { 'terra-ListItem--selected': isSelected }, { 'terra-ListItem-isSelectable': isSelectable }, { 'terra-ListItem-hasChevron': hasChevron }, className]);

  var chevron = void 0;
  if (hasChevron) {
    chevron = _react2.default.createElement(
      'span',
      { className: 'terra-ListItem-chevron' },
      _react2.default.createElement(_chevronRight2.default, { height: '10', width: '10' })
    );
  }

  return _react2.default.createElement(
    'li',
    _extends({}, customProps, { className: listItemClassNames }),
    content,
    chevron
  );
};

ListItem.propTypes = propTypes;
ListItem.defaultProps = defaultProps;

exports.default = ListItem;