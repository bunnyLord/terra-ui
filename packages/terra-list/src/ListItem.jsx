import React, { PropTypes } from 'react';
import classNames from 'classnames';
import '../src/list-item.scss';

const propTypes = {
  content: PropTypes.element,
  isSelected: PropTypes.bool,
  isSelectable: PropTypes.bool,
  hasChevron: PropTypes.bool,
};

const defaultProps = {
  content: undefined,
  isSelected: false,
  isSelectable: false,
  hasChevron: false,
};

class ListItem extends React.Component {
  static classesForListItemFromProps(isSelected, isSelectable, hasChevron) {
    return classNames(['terra-ListItem',
      { 'terra-ListItem--selected': isSelected },
      { 'terra-ListItem-selectable': isSelectable },
      { 'terra-ListItem-chevron': hasChevron },
    ]);
  }

  constructor(props) {
    super(props);
    this.onClick = this.onClick.bind(this);
  }

  render() {
    const { content, isSelected, isSelectable, hasChevron, ...customProps } = this.props;
    const listItemClasses = ListItem.classesForListItemFromProps(isSelected, isSelectable, hasChevron);
    const listItemClassNames = classNames([
      listItemClasses,
      customProps.className,
    ]);

    return (
      <div {...customProps} className={listItemClassNames}>
        {content}
      </div>
    );
  }
}

ListItem.propTypes = propTypes;
ListItem.defaultProps = defaultProps;

export default ListItem;
