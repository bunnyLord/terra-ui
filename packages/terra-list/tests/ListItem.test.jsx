import React from 'react';
import initStoryshots from 'storyshots';
import ListItem from '../src/ListItem';
// Run snapshot tests for react-storybook
initStoryshots();

// Snapshot Tests
it('should render with className', () => {
  const item = shallow(<ListItem className="textClass" />);
  expect(item).toMatchSnapshot();
});

it('should render with content', () => {
  const textContent = <p>text</p>;
  const item = shallow(<ListItem content={textContent} />);
  expect(item).toMatchSnapshot();
});

it('should render with selected', () => {
  const item = shallow(<ListItem isSelected />);
  expect(item).toMatchSnapshot();
});

it('should render with isSelectable', () => {
  const item = shallow(<ListItem isSelectable />);
  expect(item).toMatchSnapshot();
});

it('should render with isSelectable', () => {
  const item = shallow(<ListItem isSelectable hasChevron />);
  expect(item).toMatchSnapshot();
});
