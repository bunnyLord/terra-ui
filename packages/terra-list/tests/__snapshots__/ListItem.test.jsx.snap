exports[`test should render with className 1`] = `
<li
  className="terra-ListItem textClass" />
`;

exports[`test should render with content 1`] = `
<li
  className="terra-ListItem">
  <p>
    text
  </p>
</li>
`;

exports[`test should render with isSelectable 1`] = `
<li
  className="terra-ListItem terra-ListItem-isSelectable" />
`;

exports[`test should render with isSelectable and hasChevron 1`] = `
<li
  className="terra-ListItem terra-ListItem-isSelectable terra-ListItem-hasChevron">
  <span
    className="terra-ListItem-chevron">
    <mockSvg
      height="10"
      width="10" />
  </span>
</li>
`;

exports[`test should render with selected 1`] = `
<li
  className="terra-ListItem terra-ListItem--selected" />
`;
